### DISCUSSION TITLE
What is the equivalent of StartupNextInstanceEventArgs in application events in WPF?  And how can it be assigned in WPF?

### DISCUSSION THREAD

**Discussion by VahidEra** [dotnet/wpf#7517] [Q&A] [answered]
Hi guys,
I used to StartupNextInstanceEventArgs in application events option in winform to detect the execution of the next file like this:
```
Private Sub MyApplication_StartupNextInstance(sender As Object, e As ApplicationServices.StartupNextInstanceEventArgs) Handles Me.StartupNextInstance

      If e.CommandLine.Count > 0 Then
                For Each arg As String In e.CommandLine
                    'Process the commandline args
                    Dim Dir As String = arg              
                    IPlayer.AxWindowsMediaPlayer1.URL = (Dir)
                    IPlayer.OpenFileDialog1.FileName = (Dir)
                Next
       End If

End Sub
```

Now I am using WPF and there is no such thing as StartupNextInstanceEventArgs  in application events.

Of course, what came to my mind was:

```
 Private Sub MainWindow_Loaded(sender As Object, e As RoutedEventArgs) Handles Me.Loaded

   If Process.GetProcessesByName(System.Windows.Application.ResourceAssembly.GetName().Name).Length > 1 Then
         MessageBox.Show("Max 1 instance of program. Sorry.")
          System.Windows.Application.Current.Shutdown()
    End If

End Sub
```

But this is not enough because I want the previous file to stop and the new file to run like StartupNextInstanceEventArgs in application events.
>***how can i do that in WPF?
Thank you for helping me.***

**Comment by miloush** [comment, marked as answer]
There is a general .NET way, see https://learn.microsoft.com/en-us/dotnet/api/system.environment.getcommandlineargs?view=net-7.0

Otherwise in WPF this is in Application.OnStartup / Application.Startup

**Reply by VahidEra** [reply]
@miloush Dear miloush, thank you for help. This was exactly what I was looking for.
